### PR TITLE
Remove 2 failing ONNX conversion tests

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

After ##22212, two tests start to fail.

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
This is a legacy script, users are supposed to use `optimum` now.

**Comment by ydshieh** [comment]
OK. But should we do something like removing CI regarding this? Currently failing tests pop up.

**Comment by sgugger** [comment]
I think you can remove the tests as well.

**Comment by ydshieh** [comment]
@sgugger Just to confirm, we want/could remove all things like 

- `ConvertCommand` in `src/transformers/commands/transformers_cli.py`
- `export_with_transformers` in `src/transformers/commands/transformers_cli.py`
- the file `src/transformers/onnx/convert.py` and any test using this fle

also cc @michaelbenayoun @fxmarty

**Comment by sgugger** [comment]
No we're not removing code, just the tests if they start failing.

**Comment by ydshieh** [comment]
OK, glad I ask!

File: src/transformers/onnx/features.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ class FeaturesManager:
             onnx_config_cls="models.deberta_v2.DebertaV2OnnxConfig",
         ),
         "deit": supported_features_mapping(
-            "default", "image-classification", "masked-im", onnx_config_cls="models.deit.DeiTOnnxConfig"
```
Comment (ydshieh) [inline src/transformers/onnx/features.py:270]: we just need to remove the model architecture (task) that cause problem after #22212

File: src/transformers/onnx/features.py [hunk 2]
```diff
@@ -267,7 +267,7 @@ class FeaturesManager:
...
+            "default", "image-classification", onnx_config_cls="models.deit.DeiTOnnxConfig"
```
Comment (ydshieh) [inline src/transformers/onnx/features.py:270]: we just need to remove the model architecture (task) that cause problem after #22212

**Comment by ydshieh** [comment]
Ping @sgugger again to draw a bit of his attention.

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Pull request merged**